### PR TITLE
Lambdas for Remaining Modules JPA -> ZK

### PR DESCRIPTION
**Pull request by garyrussell**

### CODE REVIEW

File: spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java [hunk 1]
```diff
@@ -703,55 +577,35 @@ public void testNoInitialIdleDelayWhenRecentNotSupported() throws Exception {
...
+		doAnswer(invocation -> null).when(receiver).fetchMessages(messages);
 
-			@Override
-			public Object answer(InvocationOnMock invocation) throws Throwable {
-				Thread.sleep(5000);
-				shouldFindMessagesCounter.set(1);
-				return null;
-			}
+		doAnswer(invocation -> {
+			Thread.sleep(5000);
```
Comment (artembilan) [inline spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java]: Can we get rid of this long sleep somehow?
I understand that this is a `LongRunningIntegrationTest`, but still...
Sometime I run `testAll` locally and our CI plan for IO performs `testAll`, too. Twice.

File: spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java [hunk 2]
```diff
@@ -898,34 +720,26 @@ public void testImapLifecycleForRaceCondition() throws Exception {
...
-							e.printStackTrace();
-							failed.getAndIncrement();
-						}
+			new Thread(() -> {
+				try {
+					receiver.receive();
+				}
+				catch (javax.mail.MessagingException e) {
+					if (e.getCause() instanceof NullPointerException) {
+						e.printStackTrace();
```
Comment (artembilan) [inline spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java]: No `printStackTrace()`.

File: spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java [hunk 3]
```diff
@@ -898,34 +720,26 @@ public void testImapLifecycleForRaceCondition() throws Exception {
...
-						// ignore
-						ignore.printStackTrace();
-					}
+			new Thread(() -> {
+				try {
+					receiver.destroy();
+				}
+				catch (Exception ignore) {
+					// ignore
+					ignore.printStackTrace();
```
Comment (artembilan) [inline spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java]: DITTO

File: spring-integration-redis/src/test/java/org/springframework/integration/redis/util/AggregatorWithRedisLocksTests.java [hunk 4]
```diff
@@ -131,17 +131,13 @@ public void testLockThreeGroups() throws Exception {
...
-				catch (Exception e) {
-					e.printStackTrace();
-					exception = e;
-				}
+		Executors.newSingleThreadExecutor().execute(() -> {
+			try {
+				in2.send(new GenericMessage<String>("bar", stubHeaders(2, 2, 1)));
+			}
+			catch (Exception e) {
+				e.printStackTrace();
```
Comment (artembilan) [inline spring-integration-redis/src/test/java/org/springframework/integration/redis/util/AggregatorWithRedisLocksTests.java]: Again

File: spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpointTests.java [hunk 5]
```diff
@@ -146,15 +133,9 @@ public void testWithErrorChannel() throws NotConnectedException {
...
-			@Override
-			public void handleMessage(org.springframework.messaging.Message<?> message)
-					throws MessagingException {
-				throw new RuntimeException("ooops");
-			}
-
-		});
+		outChannel.subscribe(message -> {
+throw new RuntimeException("ooops");
+});
```
Comment (artembilan) [inline spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpointTests.java]: Something wrong with formatting.

**Review by artembilan — CHANGES_REQUESTED** [review]
The slow email tests should be reconsidered in the separate task. Here I would like do not see `printStackTrace()` and that simple code formatting issue.
Thanks

File: spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java (hunk 1 above)
Comment (garyrussell) [inline spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java]: I can't get too excited about this one test, at 5 seconds, it represents only 6% of the total run time for this class. Feel free to put a JIRA on the backlog to look at this 80 second test class.

Comment (artembilan) [inline spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java]: Here you are : https://jira.spring.io/browse/INT-4150

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/67d6cd0c89ecb6e948887fd25de6f86705eac0e9

**Pull request closed without merging**